packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -358,6 +360,7 @@ class NavigatorState extends State<Navigator> {
       if (oldRoute != null)
         oldRoute.didChangeNext(route);
       config.observer?.didPush(route, oldRoute);
+      developer.postEvent('Flutter.Route', <String, dynamic>{ 'route': route.toString() });
```

Comment:
if `route` is a `ModalRoute`, then you can get the route name from `route.settings.name`. Other routes don't have a name.